dspy/predict/predict.py

```diff
@@ -88,12 +89,8 @@ def forward(self, **kwargs):
         if new_signature is not None:
             signature = dsp.Template(signature.instructions, **new_signature)
 
-        if self.lm is None:
+        with dsp.settings.context(lm=lm):
             x, C = dsp.generate(signature, **config)(x, stage=self.stage)
-        else:
-            with dsp.settings.context(lm=self.lm, query_only=True):
```

Comment:
The only problem is that this part is necessary (`query_only=True`) for BootstrapFinetune. There's probably a cleaner way to do that but I'm not sure how.

Reply:
If we can make sure that part of BootstrapFinetune is not affected, I'll merge!

Reply:
See this and the TODO/FIXME above it: https://github.com/stanfordnlp/dspy/blob/bd82edb13492d526c0329bf0aaec9547938eb68e/dspy/teleprompt/finetune.py#L164

I think what we need is self.query_only = False in __init__ here, and basically we can set .query_only=True in teleprompt/finetune.py line 165

Reply:
I see, sorry for missing that one, there was a typo on that comment (`only_query`) and when I searched for `query_only` it did not appear on the results.